pkg/controller/volume/attachdetach/cache/actual_state_of_world.go

```diff
@@ -324,17 +327,20 @@ func (asw *actualStateOfWorld) SetVolumeMountedByNode(
 	if err != nil {
 		return fmt.Errorf("Failed to SetVolumeMountedByNode with error: %v", err)
 	}
+	// TODO: probably safer to switch getNodeAndVolume to return pointers?
 
 	if mounted {
 		// Increment set count
 		nodeObj.mountedByNodeSetCount = nodeObj.mountedByNodeSetCount + 1
 	} else {
 		// Do not allow value to be reset unless it has been set at least once
+		// TODO: not sure I understand the logic here?
 		if nodeObj.mountedByNodeSetCount == 0 {
 			return nil
 		}
 	}
 
+	// TODO-P0: volumeObj could be attachedVolume{} (if volume not found)
```

Comment:
Actually when I look at this again, I think it is not possible. Since this function is locked, the local state data structure cannot be modified during this time. If getNodeAndVolume does not return error, it means that volumeObj contains this node object. Otherwise, it will return error.